dcpy/models/product/dataset/metadata_v2.py

```diff
@@ -94,21 +93,7 @@ def _validate_colum_types(cls, v):
             assert v in get_args(COLUMN_TYPES)
         return v
 
-
-class DatasetColumn(DatasetColumnOverrides):
-    """Like DatasetColumnOverrides, but with constraints for non-null fields"""
-
-    @field_validator("name")
```

Comment:
I think these were never running before, based on the docs maybe it's something to do with these not being annotated as classmethods?

https://docs.pydantic.dev/latest/concepts/validators/#field-validators

Reply:
Thanks for wading into this. I was never quite sure how best implement nullability for the overrides. Ideally there's a shared set of fields for the class and overrides class, which suggest inheritance... but that's got some serious drawbacks. Probably has to be somewhat dynamic, which makes anything `mypy` related kind of a pain